examples/permissioned-token/test/happy-paths.js

```diff
@@ -0,0 +1,205 @@
+const { types } = require('@algo-builder/runtime');
+const { assert } = require('chai');
+const {
+  optInToASA,
+  optInToPermissions,
+  issue,
+  whitelist,
+  fund,
+  killToken,
+  transfer,
+  optOut,
+  forceTransfer,
+  setupEnv
+} = require('./common');
+
+describe('Permissioned Token Tests -- Happy Paths', function () {
+  let runtime, master, alice, bob, elon;
+  let lsig, assetIndex, controllerAppID, permissionsAppId;
+
+  // Sync Accounts and ASA def
+  function syncAccounts () {
+    master = runtime.getAccount(master.address);
+    alice = runtime.getAccount(alice.address);
+    bob = runtime.getAccount(bob.address);
+    elon = runtime.getAccount(elon.address);
+  }
+
+  this.beforeEach(async function () {
+    // Create Accounts and Env
+    [
+      runtime,
+      master,
+      alice,
+      bob,
+      elon,
+      assetIndex,
+      controllerAppID,
+      lsig,
+      permissionsAppId
+    ] = setupEnv();
+  });
+
+  it('Token Issuance', () => {
+    // Can issue after opting-in
+    optInToASA(runtime, elon.address, assetIndex);
+    syncAccounts();
+    issue(runtime, alice, elon, 20, controllerAppID, assetIndex, lsig);
+    syncAccounts();
+    assert.equal(20, runtime.getAssetHolding(assetIndex, elon.address).amount);
+  });
+
+  it('Kill Token', () => {
+    // Works as correct ASA manager is passed
+    killToken(runtime, alice, controllerAppID);
+    assert.throws(() => issue(runtime, alice, elon, 20, controllerAppID, assetIndex, lsig), 'RUNTIME_ERR1009');
+  });
+
+  it('WhiteListing', () => {
+    // Only asset manager can whitelist
+    const address = elon.address;
+    optInToPermissions(runtime, address, permissionsAppId);
+    syncAccounts();
+    assert.isDefined(elon.getAppFromLocal(permissionsAppId));
+
+    // Works with correct ASA manager
+    whitelist(runtime, alice, address, assetIndex, controllerAppID, permissionsAppId);
+    syncAccounts();
+    assert.equal(
+      Number(elon.getLocalState(permissionsAppId, 'whitelisted')),
+      1
+    );
+  });
+
+  it('Opt Out', () => {
+    // Opt-In
+    optInToASA(runtime, elon.address, assetIndex);
+    syncAccounts();
+    issue(runtime, alice, elon, 20, controllerAppID, assetIndex, lsig);
+    syncAccounts();
+    assert.equal(
+      Number(runtime.getAssetHolding(assetIndex, elon.address).amount),
+      20
```

Comment:
I think it should be:
```js
assert.equal(
  runtime.getAssetHolding(assetIndex, elon.address).amount,
  20n
);
```
Because if amount is bigger than 2^52, then current assertion would fail due to being maxed_out